src/@next/components/atoms/I18nLoader/I18nLoader.tsx

```diff
@@ -0,0 +1,45 @@
+import React from "react";
+import { I18nContext } from ".";
+import { IProps } from "./types";
+
+const fallbackLanguage = "en";
+
+const getLangCode = (code: string) =>
+  code.includes("-") ? code.split("-")[0] : code;
+
+const defaultLanguage = getLangCode(
+  (
+    (navigator.languages && navigator.languages[0]) ||
+    navigator.language ||
+    fallbackLanguage
+  ).toLowerCase()
+);
+
+export const I18nLoader: React.FC<IProps> = ({ children, languages }: IProps) => {
+  const [ language, setLanguage ] = React.useState<string>(() => {
+    return languages.hasOwnProperty(defaultLanguage) ? defaultLanguage : fallbackLanguage
+  });
+  const [ catalogs, setCatalogs ] = React.useState<any>({});
+
+  React.useEffect(() => {
+    const loadCatalog = async (language:string) => {
+      const catalog = await import(
```

Comment:
maybe we should wrap it in try catch block, because when selected language doesn't have messages, app crashes.

Reply:
We have introduced the list of available languages to make sure that we are able to load only existing translations. In addition, we will use it to build language switcher.